details/impersonation.md

```diff
@@ -0,0 +1,144 @@
+
+How does ROR use the mocks? Let's suppose we have `ldap_auth` rule. When ROR processes the rule, it:
+* asks a given LDAP if the username can be authenticated with a given password, and if they can ...
+* asks LDAP to list what groups the user belongs to
+
+In the impersonation case, it looks pretty much the same. Just we won't call any LDAP server, rather, the mock will provide the required information (no passwords required):
+* asks the mock if the username exists, and if it does ...
+* asks the mock to tell what groups the user belongs to
+
+**⚠️ IMPORTANT:** When some external service is not mocked, ROR might return to Kibana information that the impersonation is not supported. It's better to always define all mocks, to avoid the "not supported impersonation" ES response.
+
+How to define mocks in Kibana is described [here](../examples/impersonation/external-services-mocks-ui.md).
+
+#### Impersonating a chosen user
+
+Now, when we have configured Test Settings and External Services Mocks, we can try to impersonate a user. In ROR Settings, user can be:
+* provided statically (defined in the settings),
+* provided dynamically:
+   * from external, dependant systems (like LDAP) - we mock them
+   * from upstream systems (eg. through headers) - they are unknown
+
+It means that we pick the users defined in Settings or Mocks, but also we can enter the username and try to impersonate such user.
+
+[Here](../examples/impersonation/impersonate-user-ui.md) you find a description how to impersonate a user in ROR Kibana.
+
+## Logs & audit
+
+In ES logs in `USR` admin is going to find sth like this: `admin1 as (user1)` - it means that `admin1` was authenticated and they are the impersonator who is impersonating `user1`. 
+
+All logs of impersonated user in Kibana will have this format `[<log level>][plugins][ReadonlyREST][<filename>][impersonating <impersonated user username>]`
+
+When auditing is enabled, the audit document is going to contain an `impersonated_by` field.
+
+## Impersonation limitations
+
+Impersonation mode has some limitations. Please check if they have an impact on your use cases:
+
+* Not all features available in the ROR configuration are testable with impersonation mode.
+Some rules used in ROR ACL do not support impersonation. For example, auth rule with hashed credentials(e.g. `auth_key_sha512`) can be used in impersonation mode only when credentials follow the format `USER_NAME: HASH(PASSWORD)`; A fully hashed username and password don't allow fetching a username. The auth rule in such a format won't match during impersonation.
+A list of the rules with info about impersonation support can be found [here](../elasticsearch.md#rules).
+
+* Test settings are stored in the memory of the node that handled the POST tests settings request.
```

Comment:
- Test Settings should be capitalised all through the document, for consistency's sake
- `...handled the POST Test Settings request`